libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/dialogs/TextFieldDialog.kt

```diff
@@ -0,0 +1,154 @@
+import io.element.android.compound.theme.ElementTheme
+import io.element.android.libraries.designsystem.components.list.TextFieldListItem
+import io.element.android.libraries.designsystem.preview.ElementPreview
+import io.element.android.libraries.designsystem.preview.PreviewsDayNight
+import io.element.android.libraries.designsystem.theme.components.Text
+import io.element.android.libraries.ui.strings.CommonStrings
+
+@Composable
+fun TextFieldDialog(
+    title: String,
+    onSubmit: (String) -> Unit,
+    onDismissRequest: () -> Unit,
+    value: String?,
+    placeholder: String?,
+    modifier: Modifier = Modifier,
+    validation: (String?) -> Boolean = { true },
+    onValidationErrorMessage: String? = null,
+    autoSelectOnDisplay: Boolean = true,
+    maxLines: Int = 1,
+    content: String? = null,
+    label: String? = null,
+    withBorder: Boolean = false,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    submitText: String = stringResource(CommonStrings.action_ok),
+) {
+    val focusRequester = remember { FocusRequester() }
+    var textFieldContents by rememberSaveable(stateSaver = TextFieldValue.Saver) {
+        mutableStateOf(
+            TextFieldValue(
+                value.orEmpty(),
+                selection = TextRange(value.orEmpty().length)
+            )
+        )
+    }
+    var error by rememberSaveable { mutableStateOf(if (!validation(value.orEmpty())) onValidationErrorMessage else null) }
+    var canRequestFocus by rememberSaveable { mutableStateOf(false) }
+    val canSubmit by remember { derivedStateOf { validation(textFieldContents.text) } }
+    ListDialog(
+        title = title,
+        onSubmit = { onSubmit(textFieldContents.text) },
+        onDismissRequest = onDismissRequest,
+        enabled = canSubmit,
+        applyPaddingToContents = content.isNullOrEmpty().not(),
+        submitText = submitText,
+        modifier = modifier,
+    ) {
+        if (content != null) {
+            item {
+                Text(
+                    text = content,
+                    style = ElementTheme.materialTypography.bodyMedium,
+                )
+            }
+        }
+        item {
+            TextFieldListItem(
+                placeholder = placeholder.orEmpty(),
+                label = label,
+                withBorder = withBorder,
+                text = textFieldContents,
+                onTextChange = {
+                    error = if (!validation(it.text)) onValidationErrorMessage else null
+                    textFieldContents = it
+                },
+                error = error,
+                keyboardOptions = keyboardOptions,
+                keyboardActions = KeyboardActions(onAny = {
+                    if (validation(textFieldContents.text)) {
+                        onSubmit(textFieldContents.text)
+                    }
+                }),
+                maxLines = maxLines,
+                modifier = Modifier.focusRequester(focusRequester),
```

Comment:
Actually, shouldn't this be `Modifier.fillMaxWidth().focus...` ? I just tried replacing the EC base URL component with this dialog and the text field doesn't reach the end.

Reply:
Yes, maybe. Not that this code is not new. I'll check it and add more commit

Reply:
Actually, this PR is touching unrelated code (room member moderation), let's handle that in a dedicated one.